modules/integration/src/test/scala/scala/cli/integration/RunPipedSourcesTestDefinitions.scala

```diff
@@ -202,4 +202,14 @@ trait RunPipedSourcesTestDefinitions { _: RunTestDefinitions =>
       expect(output == expectedOutput)
     }
   }
+
+  test("running script snipped should not create workspace dir in cwd") {
+    emptyInputs.fromRoot { root =>
+      os.proc(TestUtil.cli, "-e", """println("Hello")""", extraOptions)
+        .call(cwd = root)
+        .out.trim()
+      expect(!os.exists(root / Constants.workspaceDirName))
+    }
+  }
```

Comment:
Also, we could have another test confirming the workspace directory **does** get created in the project folder for mixed virtual and on-disk sources, just to be extra sure.